presto-main/src/main/java/com/facebook/presto/operator/aggregation/arrayagg/SetAggregationFunction.java

```diff
@@ -13,124 +13,32 @@
-        ArrayType outputType = (ArrayType) functionAndTypeManager.getParameterizedType(StandardTypes.ARRAY, ImmutableList.of(
-                TypeSignatureParameter.of(type.getTypeSignature())));
-        return generateAggregation(type, outputType);
-    }
-
-    @Override
-    public boolean isCalledOnNullInput()
-    {
-        return true;
-    }
-
-    private static BuiltInAggregationFunctionImplementation generateAggregation(Type type, ArrayType outputType)
-    {
-        DynamicClassLoader classLoader = new DynamicClassLoader(SetAggregationFunction.class.getClassLoader());
-
-        List<Type> inputTypes = ImmutableList.of(type);
-        AccumulatorStateSerializer<?> stateSerializer = new SetAggregationStateSerializer(outputType);
-        AccumulatorStateFactory<?> stateFactory = new SetAggregationStateFactory(type);
-
-        Type intermediateType = stateSerializer.getSerializedType();
-        List<ParameterMetadata> inputParameterMetadata = createInputParameterMetadata(type);
-        Class<? extends AccumulatorState> stateInterface = SetAggregationState.class;
-        AggregationMetadata metadata = new AggregationMetadata(
-                generateAggregationName(NAME, outputType.getTypeSignature(), inputTypes.stream().map(Type::getTypeSignature).collect(toImmutableList())),
-                inputParameterMetadata,
-                INPUT_FUNCTION.bindTo(type),
-                COMBINE_FUNCTION,
-                OUTPUT_FUNCTION,
-                ImmutableList.of(new AccumulatorStateDescriptor(
-                        stateInterface,
-                        stateSerializer,
-                        stateFactory)),
-                outputType);
-
-        Class<? extends Accumulator> accumulatorClass = AccumulatorCompiler.generateAccumulatorClass(
-                Accumulator.class,
-                metadata,
-                classLoader);
-        Class<? extends GroupedAccumulator> groupedAccumulatorClass = AccumulatorCompiler.generateAccumulatorClass(
-                GroupedAccumulator.class,
-                metadata,
-                classLoader);
-        return new BuiltInAggregationFunctionImplementation(NAME, inputTypes, ImmutableList.of(intermediateType),
-                outputType, true, true, metadata, accumulatorClass, groupedAccumulatorClass);
-    }
-
-    private static List<ParameterMetadata> createInputParameterMetadata(Type value)
+    private SetAggregationFunction()
     {
-        return ImmutableList.of(new ParameterMetadata(STATE), new ParameterMetadata(NULLABLE_BLOCK_INPUT_CHANNEL, value), new ParameterMetadata(BLOCK_INDEX));
     }
 
-    public static void input(Type type, SetAggregationState state, Block block, int position)
+    @InputFunction
+    @TypeParameter("T")
+    public static void input(@TypeParameter("T") Type type, @AggregationState SetAggregationState state, @BlockPosition @SqlType("T") @NullablePosition Block block, @BlockIndex int position)
```

Comment:
nit: For readability, I'd recommend breaking this up by putting each argument on its own line.